common/elasticsearch/src/main/scala/uk/ac/wellcome/elasticsearch/IndexConfig.scala

```diff
@@ -14,6 +14,7 @@ import uk.ac.wellcome.elasticsearch.WorksAnalysis._
 trait IndexConfig {
   def mapping: MappingDefinition
   def analysis: Analysis
+  def shards: Int = 1
```

Comment:
Is there a reason this is `def` rather than `val`?

Reply:
Not really. I never remember variables initialization order when they're defined in a trait and overridden in extending classes. Using `def` just feels a little bit safer cause I don't need to know that order since a def is always re-evaluated when called. Where it's overridden in `WorksIndexConfig` it's defined as a val

Reply:
Also in general making things `val` in the base interface over constrains things (can't make dynamic in the implementer if required, always has to be fixed at creation time)